src/main/java/org/quiltmc/asmr/processor/AsmrProcessor.java

```diff
@@ -196,16 +196,16 @@ public void process() {
 
         // compute rounds transformer action
         currentAction = AsmrTransformerAction.COMPUTE_ROUNDS;
-        List<List<AsmrTransformer>> rounds = computeRounds();
+        List<? extends List<? extends AsmrTransformer>> rounds = computeRounds();
```

Comment:
```suggestion
        List<List<AsmrTransformer>> rounds = computeRounds();
```
A `List<? extends Foo>` is not a list of something that can contain subclasses of foo, it's any list that contains *only* subclasses of `Foo`. For example, a `List<Bar>` can be assigned to a `List<? extends Foo>` if `Bar` extends `Foo`, but a `Bar` can safely be added to a `List<Foo>`.
There's a lot of examples like this throughout this PR, please revert those and let me know whether that affects the unchecked warning.